tests/api_test.py

```diff
@@ -1093,6 +1093,42 @@ def vjp(x_tangent):
     b = np.dot(a + np.eye(a.shape[0]), real_x)
     print(gf(a, b))  # doesn't crash
 
+  def test_vmap_in_axes_tree_prefix_error(self):
+    # https://github.com/google/jax/issues/795
+    jtu.check_raises_regexp(
+        lambda: api.vmap(lambda x: x, in_axes=(0, 0))(np.ones(3)),
+        ValueError, "axes specification must be a tree prefix")
```

Comment:
nit: I like to put the whole error message so you/I/future readers can quickly check that the error message is clear, correct, etc.

Reply:
Done.